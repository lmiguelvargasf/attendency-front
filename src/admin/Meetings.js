import React from 'react'
import useAxios from 'axios-hooks'
import { Table, Space } from 'antd'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEdit, faTrash, faTimes } from '@fortawesome/free-solid-svg-icons'

const Meetings = () => {
  const [{ data: meetings, loading, error }] = useAxios(
    `${process.env.REACT_APP_API_URL}/meetings/`
  )

  if (loading) return <p data-testid='loading'>Loading...</p>
  if (error) return <p data-testid='error'>Error!</p>

  const columns = [
    {
      title: 'Project',
      dataIndex: 'project',
      key: 'project',
      render: project => <a>{project}</a>
    },
    {
      title: 'Date',
      dataIndex: 'date',
      key: 'date'
    },
    {
      title: 'Time',
      dataIndex: 'time',
      key: 'time'
    },
    {
      title: 'Action',
      key: 'action',
      render: (text, record) => (
        <Space size='middle'>
          <FontAwesomeIcon icon={faEdit} />
          <FontAwesomeIcon icon={faTrash} />
          <FontAwesomeIcon icon={faTimes} />
        </Space>
      )
    }
  ]

  return (
    <Table
      data-testid='meeting-table'
      columns={columns}
      dataSource={meetings}
      pagination={false}
    />
  )
}

export default Meetings
