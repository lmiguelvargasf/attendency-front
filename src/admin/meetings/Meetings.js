import React, { useEffect, useState } from 'react'
import { Route, Switch } from 'react-router-dom'
import useAxios from 'axios-hooks'
import MeetingsTable from './MeetingsTable'
import CreateMeeting from './CreateMeeting'
import EditMeeting from './EditMeeting'

const Meetings = () => {
  const [meetings, setMeetings] = useState([])
  const [{ data, loading, error }] = useAxios(
    `${process.env.REACT_APP_API_URL}/meeting-table/`
  )

  useEffect(() => { setMeetings(data) }, [data])

  const removeMeeting = meetingToDelete => {
    setMeetings(() => meetings.filter(meeting => meeting.key !== meetingToDelete.key))
  }
  const addMeeting = meeting => {
    setMeetings(meetings => [meeting, ...meetings])
  }
  const updateMeetings = (updatedMeeting, index) => {
    meetings[index] = updatedMeeting
    setMeetings(meetings)
  }

  if (loading) return <p data-testid='loading'>Loading...</p>
  if (error) return <p data-testid='error'>Error!</p>

  return (
    <Switch>
      <Route path='/admin/meetings/create' render={props => <CreateMeeting {...props} addMeeting={addMeeting} />} />
      <Route path='/admin/meetings/:key/edit' render={props => <EditMeeting {...props} updateMeetings={updateMeetings} />} />
      <Route path='/admin/meetings' render={props => <MeetingsTable {...props} meetings={meetings} removeMeeting={removeMeeting} />} />
    </Switch>
  )
}

export default Meetings
