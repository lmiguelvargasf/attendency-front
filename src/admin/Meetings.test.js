import React from 'react'
import renderer from 'react-test-renderer'
import Meetings from './Meetings'

it('renders Meetings component properly', () => {
  const tree = renderer.create(<Meetings />)
  expect(tree).toMatchSnapshot()
})
