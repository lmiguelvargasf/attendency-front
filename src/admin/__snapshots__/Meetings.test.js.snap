// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Meetings component properly 1`] = `
<p>
  Loading...
</p>
`;
